app/src/main/java/com/example/capstone/ProfileCreationFragment.java

```diff
@@ -22,12 +65,119 @@ public static ProfileCreationFragment newInstance(String param1, String param2)
+                                    ParseFile profilePicture = new ParseFile("profile_pic", profilePictureByteArr);
+                                    user.put("profile_picture", profilePicture);
+                                }
+                                user.saveInBackground(new SaveCallback() {
+                                    @Override
+                                    public void done(ParseException e) {
+                                        if (e == null) {
+                                            Toast.makeText(getActivity(), "Saved", Toast.LENGTH_SHORT).show();
+                                        } else {
+                                            Toast.makeText(getActivity(), "Failed to Save", Toast.LENGTH_SHORT).show();
+                                            Log.d(getClass().getSimpleName(), "User update error: " + e);
+                                        }
+                                    }
+                                });
+                            } else {
+                                e.printStackTrace();
+                            }
+                        }
+                    });
+
+                    Intent home = new Intent(getActivity(), HomescreenActivity.class);
+                    startActivity(home);
+                }
+            }
+        });
+    }
+
+    public void onPickPhoto(View view) {
+        Intent intent = new Intent(Intent.ACTION_PICK,
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+            startActivityForResult(intent, PICK_PHOTO_CODE);
+    }
+
+    public Bitmap loadFromUri(Uri photoUri) {
+        Bitmap image = null;
+        try {
+            if(Build.VERSION.SDK_INT > 27){
+                // on newer versions of Android, use the new decodeBitmap method
+                ImageDecoder.Source source = ImageDecoder.createSource(getActivity().getContentResolver(), photoUri);
+                image = ImageDecoder.decodeBitmap(source);
+            } else {
+                // support older versions of Android by using getBitmap
+                image = MediaStore.Images.Media.getBitmap(getActivity().getContentResolver(), photoUri);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return image;
+    }
+
+    private static byte[] bitmapToByteArr(Bitmap bitmap){
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+        byte[] byteArray = stream.toByteArray();
+        return byteArray;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if ((data != null) && requestCode == PICK_PHOTO_CODE) {
+            Uri photoUri = data.getData();
+            Bitmap selectedImage = loadFromUri(photoUri);
+            ivProfilePicture.setImageBitmap(selectedImage);
+            profilePictureByteArr = bitmapToByteArr(selectedImage);
```

Comment:
This is very well written. I can follow the logic easily here even though there are no comments.

Two points though:
1. Use `Array` instead of `Arr`. We want to use clear verbose names rather than short hands and comments.
2. The `profilePictureByteArr` result is not used anywhere?